pyomo/contrib/doe/examples/fim_doe_tutorial.ipynb

```diff
@@ -57,77 +137,340 @@
+    "            if t in m.t_con:\n",
+    "                return pyo.Constraint.Skip\n",
+    "            else:\n",
+    "                neighbour_t = max(tc for tc in control_time if tc < t)\n",
+    "                return m.T[t] == m.T[neighbour_t]\n",
+    "\n",
+    "        def cal_kp1(m, t):\n",
+    "            \"\"\"\n",
+    "            Create the perturbation parameter sets\n",
+    "            m: model\n",
+    "            t: time\n",
+    "            \"\"\"\n",
+    "            # LHS: 1/h\n",
+    "            # RHS: 1/h*(kJ/mol *1000J/kJ / (J/mol/K) / K)\n",
+    "            return m.kp1[t] == m.A1 * pyo.exp(-m.E1 * 1000 / (m.R * m.T[t]))\n",
+    "\n",
+    "        def cal_kp2(m, t):\n",
+    "            \"\"\"\n",
+    "            Create the perturbation parameter sets\n",
+    "            m: model\n",
+    "            t: time\n",
+    "            \"\"\"\n",
+    "            # LHS: 1/h\n",
+    "            # RHS: 1/h*(kJ/mol *1000J/kJ / (J/mol/K) / K)\n",
+    "            return m.kp2[t] == m.A2 * pyo.exp(-m.E2 * 1000 / (m.R * m.T[t]))\n",
+    "\n",
+    "        def dCdt_control(m, y, t):\n",
+    "            \"\"\"\n",
+    "            Calculate CA in Jacobian matrix analytically\n",
+    "            y: CA, CB, CC\n",
+    "            t: timepoints\n",
+    "            \"\"\"\n",
+    "            if y == \"CA\":\n",
+    "                return m.dCdt[y, t] == -m.kp1[t] * m.C[\"CA\", t]\n",
+    "            elif y == \"CB\":\n",
+    "                return m.dCdt[y, t] == m.kp1[t] * m.C[\"CA\", t] - m.kp2[t] * m.C[\"CB\", t]\n",
+    "            elif y == \"CC\":\n",
+    "                return pyo.Constraint.Skip\n",
+    "\n",
+    "        def alge(m, t):\n",
+    "            \"\"\"\n",
+    "            The algebraic equation for mole balance\n",
+    "            z: m.pert\n",
+    "            t: time\n",
+    "            \"\"\"\n",
+    "            return m.C[\"CA\", t] + m.C[\"CB\", t] + m.C[\"CC\", t] == m.CA0[0]\n",
+    "\n",
+    "        # Control time\n",
+    "        mod.T_rule = pyo.Constraint(mod.t, rule=T_control)\n",
+    "\n",
+    "        # Calculating C, Jacobian, FIM\n",
+    "        mod.k1_pert_rule = pyo.Constraint(mod.t, rule=cal_kp1)\n",
+    "        mod.k2_pert_rule = pyo.Constraint(mod.t, rule=cal_kp2)\n",
+    "        mod.dCdt_rule = pyo.Constraint(mod.C_set, mod.t, rule=dCdt_control)\n",
+    "\n",
+    "        mod.alge_rule = pyo.Constraint(mod.t, rule=alge)\n",
+    "\n",
+    "        # Boundary conditions\n",
+    "        mod.C[\"CB\", 0.0].fix(0.0)\n",
+    "        mod.C[\"CC\", 0.0].fix(0.0)\n",
+    "\n",
+    "        if return_m:\n",
+    "            return mod"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
    "metadata": {
-    "scrolled": true
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# from pyomo.contrib.doe.examples.reactor_kinetics import create_model, disc_for_measure"
```

Comment:
Please add before this comment: "# The above models are alternately available in the examples folder:"